src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseService.java

```diff
@@ -951,31 +954,40 @@ public ProgrammingExercise save(ProgrammingExercise programmingExercise) {
      */
     public SearchResultPageDTO<ProgrammingExercise> getAllOnPageWithSize(final PageableSearchDTO<String> search, final Boolean isCourseFilter, final Boolean isExamFilter,
```

Comment:
```suggestion
    public SearchResultPageDTO<ProgrammingExercise> getAllOnPageWithSize(final PageableSearchDTO<String> search, final boolean isCourseFilter, final boolean isExamFilter,
```
Can you change the parameter types here, or would that require some refactoring in other places?
Applies to equivalent method on services for other exercise types as well.

Reply:
They are annotated with `@RequestParam(defaultValue = "true")` so null should not be possible.